### PR TITLE
HEEDLS-537 Common code for 532 and 416

### PR DESCRIPTION
**Pull request by AlexJacksonDS**

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/UserDataService.cs [hunk 1]
```diff
@@ -399,7 +400,8 @@ public int GetDelegateCountWithAnswerForPrompt(int centreId, int promptNumber)
                         FROM Candidates
                         WHERE CentreID = @centreId AND Answer{promptNumber} IS NOT NULL",
                 new { centreId }
-            ).Count(x => !string.IsNullOrWhiteSpace(x)); ;
+            ).Count(x => !string.IsNullOrWhiteSpace(x));
+            ;
```
Comment (stellake) [inline DigitalLearningSolutions.Data/DataServices/UserDataService.cs:404]: Is this needed?

File: DigitalLearningSolutions.Data/Models/User/AdminUser.cs [hunk 2]
```diff
@@ -33,6 +35,11 @@ public class AdminUser : User
         public bool IsLocalWorkforceManager { get; set; }
         public bool ImportOnly { get; set; }
 
+        public int FailedLoginCount { get; set; }
+
+        public bool IsLocked => FailedLoginCount >= FailedLoginThreshold;
+        public bool IsCmsAdministrator => ImportOnly && IsContentManager;
```
Comment (stellake) [inline DigitalLearningSolutions.Data/Models/User/AdminUser.cs:41]: Nice!

File: DigitalLearningSolutions.Data/DataServices/UserDataService.cs [hunk 3]
```diff
@@ -138,25 +136,28 @@ public List<AdminUser> GetAdminUsersByCentreId(int centreId)
...
+                        au.SummaryReports,
+                        au.UserAdmin AS IsUserAdmin,
+                        au.CategoryID,
+                        cc.CategoryName,
+                        au.Supervisor AS IsSupervisor,
+                        au.Trainer AS IsTrainer,
+                        au.ImportOnly,
+                        au.FailedLoginCount
+                    FROM AdminUsers AS au
+                    LEFT JOIN CourseCategories AS cc ON cc.CourseCategoryID = au.CategoryID
```
Comment (stellake) [inline DigitalLearningSolutions.Data/DataServices/UserDataService.cs:159]: Should we add/update any data tests as part of this change?

File: DigitalLearningSolutions.Data/DataServices/UserDataService.cs (hunk 1 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/DataServices/UserDataService.cs:404]: Looks like something that has been around for a while that only just got picked up, will remove

**Review by stellake — APPROVED** [review]
Looks good! 👍

**Pull request merged**